src/test_solvers/main.cpp

```diff
@@ -0,0 +1,32 @@
+#include "solvers.h"
+int main(){
+
+    // Initialize triplets for the matrix
+    std::vector<int> rowInd;
+    std::vector<int> colInd;
+    std::vector<float> val;
+    rowInd.push_back(0);
+    rowInd.push_back(1);
+    rowInd.push_back(1);
+    colInd.push_back(1);
+    colInd.push_back(0);
+    colInd.push_back(1);
+    val.push_back(1);
+    val.push_back(2);
+    val.push_back(3);
+
+    std::vector<float> b;
+    std::vector<float> x;
+    
+    b.push_back(1.7);
+    b.push_back(8.3);
+
+    x.resize(b.size());
+    // test the iterative CUSP solver
+    x = cuspSolver(&rowInd, &colInd, &val, &b);
```

Comment:
If you assign `x` here with a new value, the `x.resize()` before doesn't really make a difference.

Rather, I'd proabbly assign it directly to a const std::vector to clarify that this is a result (that then this is also not modified after).

```c++
const std::vector<float> x = cuspSolver(...);
```